src/Spago/Build.hs

```diff
@@ -200,33 +202,117 @@ run
 run maybeModuleName buildOpts extraArgs = do
   Config.Config { alternateBackend } <- view (the @Config)
   let moduleName = fromMaybe (ModuleName "Main") maybeModuleName
-  runBackend alternateBackend moduleName Nothing "Running failed; " buildOpts extraArgs
+  runBackend alternateBackend Nothing moduleName Nothing "Running failed; " buildOpts extraArgs
 
 
+-- | Run the select module as a script: init, compile, and run the provided module
+script
+  :: (HasEnv env)
+  => Text
+  -> Maybe Text
+  -> [PackageName]
+  -> RIO env ()
+script modulePath tag packageDeps = do
+  logDebug "Running `spago script`"
+  absoluteModulePath <- fmap Text.pack (makeAbsolute (Text.unpack modulePath))
+  moduleContents <- readTextFile (pathFromText absoluteModulePath)
+
+  moduleName <- case parseModuleName moduleContents of
+    Nothing -> die [ "Failed to extract module name from source file" ]
+    Just modName -> pure modName
+
+  GlobalCache cacheDir _ <- view (the @GlobalCache)
+  currentDir <- Turtle.pwd
+  Temp.withTempDirectory cacheDir "spago-script-tmp" $ \dir -> do
+    let tmpDir = Text.pack dir
+    Turtle.cd (Turtle.fromText tmpDir)
+
+    config@Config { packageSet = PackageSet{..}, ..}
+      <- Packages.initProject NoForce Dhall.WithComments tag
```

Comment:
`initProject` creates a `spago.dhall`, `packages.dhall`, `src/Main.purs`, `test/Main.purs`, `.purs-repl`, `.gitignore`, and other files as part of creating a new project. However, the `repl` command (when used outside a project) and the `script` command don't need to create any of these files. The package sets tag and the config can be handled in memory (in fact, once we read from the config for the first time we do manipulate it in memory only -- see `newConfig` below).

The `src/Main.purs` file in particular causes problems for the script command. As @colinwahl noted below, this module is included in the config sources by default and has the module name `Main`, which is very likely to cause a duplicate module compilation error with user scripts (which are also likely to be named `Main`). For this reason we edit the config (below) to strip out the sources so these files are not included in compilation.

But it seems even better to just not generate any of these files in the case we are doing a temporary project (`spago repl` outside a project or `spago script` anywhere). We didn't attempt this in this PR because it's a far-reaching change, but I wanted to suggest that we introduce some kind of `Temporary | Permanent` flag that can be used to control whether we write these files in the first place.

Reply:
I was working on this, and things are straightforward except I am not sure how to get a `PackageSet` from the tag in memory.

I think I know how to construct the URL for a specific tag, and I know how to fetch the latest tag from github (when `tag = Nothing`), but I don't know how to take a `tag :: Text` and get a `PackageSet`. It looks like all the other code paths rely on there being some `packages.dhall` file in memory that they modify.

I think that I will need to adapt some functions work on a Dhall expr in memory, then use something similar to `Spago.RunEnv.getPackageSet` to get out a `PackageSet`.

Reply:
@colinwahl maybe something like this could work?

```haskell
Dhall.inputExpr "https://raw.githubusercontent/...etc.." >>= Spago.Config.parsePackageSet
```

Reply:
Yeah, something like that that works!